### PR TITLE
HWKALERTS-88 Better integrate Alert.trigger and remove redundancies

### PR DESCRIPTION
**Pull request by jshaughn**
- remove Alert.triggerId and Alert.context.
  - redundant with Alert.trigger.id and Alert.trigger.context
- maintain some pass-thru convenience getters
- update constructors
- maintain Alert.severity
  - it is lighweight and may be modifiable (in the future)
  - still a candidate to be removed as redundant

Other:
- Fix a build issue, it seemed there was a dep issue in action-plugins

### CODE REVIEW

File: hawkular-alerts-api/src/main/java/org/hawkular/alerts/api/model/condition/Alert.java [hunk 1]
```diff
@@ -112,27 +109,20 @@
...
-        this.triggerId = triggerId;
+        this.trigger = trigger;
         this.severity = (null == severity) ? Severity.MEDIUM : severity;
         this.evalSets = evalSets;
 
         this.ctime = System.currentTimeMillis();
         this.status = Status.OPEN;
 
-        this.alertId = tenantId + "-" + triggerId + "-" + ctime;
+        this.alertId = tenantId + "-" + trigger.getId() + "-" + ctime;
```
Comment (lucasponce) [inline hawkular-alerts-api/src/main/java/org/hawkular/alerts/api/model/condition/Alert.java]: Perhaps some validation that trigger == null is not permitted ? an IllegalArgumentException sounds good.

**Comment by lucasponce** [comment]
I've put a small comment.
This PR requires changes in the UI as well, as alert.triggerId is used on several places.

As the previous one, I suggest to merge them on the new branch Events, at least until Hawkular MS5 is out and no last minute fixes are needed.

File: hawkular-alerts-api/src/main/java/org/hawkular/alerts/api/model/condition/Alert.java (hunk 1 above)
Comment (jshaughn) [inline hawkular-alerts-api/src/main/java/org/hawkular/alerts/api/model/condition/Alert.java]: I'll change this in the forthcoming events PR that is the refactor I showed you earlier today, plus the revisions resulting from our discussion (that's a US discussion, not a Spanish discussion :) )

**Comment by jshaughn** [comment]
Closing and re-opening against Events branch

**Pull request closed without merging**